modules/fm2.cpp

```diff
@@ -0,0 +1,57 @@
+#include "fm2.h"
+
+using namespace daisysp;
+
+void Fm2::Init(float samplerate)
+{
+    //init oscillators
+    car_.Init(samplerate);
+    mod_.Init(samplerate);
+
+    //set some reasonable values
+    SetFrequency(440.f);
+    SetRatio(.5f);
+
+    car_.SetAmp(1.f);
+    mod_.SetAmp(1.f);
+
+    car_.SetWaveform(Oscillator::WAVE_SIN);
+    mod_.SetWaveform(Oscillator::WAVE_SIN);
+
+    idx_ = 1.f;
+}
+
+float Fm2::Process()
+{
+    float modval = mod_.Process();
+    car_.PhaseAdd(modval * idx_);
+    return car_.Process();
+}
+
+void Fm2::SetFrequency(float freq)
+{
+    freq = fabsf(freq);
+    car_.SetFreq(freq);
+    mod_.SetFreq(freq * ratio_);
+}
+
+void Fm2::SetRatio(float ratio)
+{
+    ratio_ = fabsf(ratio);
+}
```

Comment:
With how everything else is set up, this won't adjust the modulator frequency until the next time that SetFrequency is called.

This should either do a check, and update that frequency here as well, or moving the freq recalc to `Process`

with something like:
```c++
if (lratio_ != ratio_ || lfreq_ != freq_)
{
    lratio_ = ratio_;
    lfreq_ = freq_;
    car_.SetFreq(lfreq);
    mod_.SetFreq(lfreq * lratio_);
}
```

You could use the freq getter instead of adding `lfreq_`, but will still want to use 

Leaving the stuff in the setter would also work, but when setting both Ratio and Frequency every loop (or every sample) would add an extra few multiplies for calling SetFrequency. So the local variable comparison check for updating it as infrequently as possible seems like the most efficient approach.